xds/internal/client/client_callback.go

```diff
@@ -18,22 +18,46 @@
 
 package client
 
+import "fmt"
+
 type watcherInfoWithUpdate struct {
 	wi     *watchInfo
 	update interface{}
 	err    error
 }
 
 func (c *Client) scheduleCallback(wi *watchInfo, update interface{}, err error) {
+	if err != errCallbackTimeout {
+		// Only stop the timer if this is not a timeout error (it can be either
+		// nil, or some other error).
+		wi.expiryTimer.Stop()
+	}
 	c.updateCh.Put(&watcherInfoWithUpdate{
 		wi:     wi,
 		update: update,
 		err:    err,
 	})
 }
 
+var errCallbackTimeout = fmt.Errorf("callback timeout")
```

Comment:
Prefer using `errors.New` since there are no formatting directives.

Reply:
This error is deleted.